### PR TITLE
Feat(test): Local testing environment for SAML authentication

### PR DESCRIPTION
**Pull request by Millefeuille42**
## Checkpoints
https://github.com/tracim/tracim/issues/6206

Relevant files: `tools_docker/shibboleth_idp/README.md` and `tools_docker/shibboleth_idp/Dockerfile`

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

**Comment by Millefeuille42** [comment]
@grignards is this enough for now ? I haven't edited elements such as `/tools_docker/test_docker` or `doc/TESTING.md`, because I am not sure these are relevant for now. As we don't really implemented the usage with tracim yet.

**Comment by grignards** [comment]
> @grignards is this enough for now ? I haven't edited elements such as `/tools_docker/test_docker` or `doc/TESTING.md`, because I am not sure these are relevant for now. As we don't really implemented the usage with tracim yet.

Yes, seems enough in my opinion.

**Review by grignards — APPROVED** [review]

File: tools_docker/shibboleth_idp/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Configuration
+
+[//]: # FIXME - M.L - 2023/08/30
+
+Configuration is set at build time, to apply a new configuration the image must be re-built:
+```sh
+docker build --no-cache -t shibboleth-idp-test-image .
+```
+
+To add a SP to shibboleth's know SPs, add the following line at the end of `shibboleth-idp/conf/metadata-providers.xml` before the closing `</MetadataProvider>` tag:
```
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply tools_docker/shibboleth_idp/README.md]: Done !

**Comment by Millefeuille42** [comment]
@lebouquetin Considering that this PR is from https://github.com/tracim/tracim/pull/6232, should the merge target be feat/6231__implement_saml_authentication_flow or develop ?

In the latter, this means https://github.com/tracim/tracim/pull/6232 needs to be merged in develop beforehand despite missing crucial elements implemented in this PR.

The first element should make overall development easier in my opinion, since it will permit to review each development individually without having to merge unclean code into develop

and in the end for the commit history of develop we would have something like "feat: added SAML as an authentication source", which will make it cleaner.

File: tools_docker/shibboleth_idp/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Make sure to have port 8080 available, then run the container: 
+```sh
+docker run --rm -d --name="shibboleth-idp-test-container" --network=host shibboleth-idp-test-image
+```
+
+Shibboleth is now accessible at `http://localhost:8080`, and it's metadata URL (entityID) is `http://localhost:8080/idp/shibboleth`
+
+## Caveats
+
+[//]: # FIXME - M.L - 2023/08/30
```
Comment (lebouquetin) [inline tools_docker/shibboleth_idp/README.md:31]: What is this FIXME for?

- a FIXME should come with some explanation.
- FIXMEs are not intended for merge. Why do you still have this FIXME? Should we replace it with a TODO? Fix it ?

**Review by lebouquetin — COMMENTED** [review]
@Millefeuille42 I'm not sure to understand the PR content. What I understand is:

- the PR includes an IdP which can be executed as a docker container
- the IdP is preconfigured for being ready-to-run with tracim
- the IdP is a java-based process

Questions:

- Do I understand correctly?
- Is there a documentation which help to configure & run Tracim and IdP together?

**Comment by Millefeuille42** [comment]
@lebouquetin Yes you understood correctly.

No process of configuration and testing with tracim has been done yet, since SAML wasn't in tracim at the moment.

Documentation about this should be done afterwards (for example in #6232)

File: tools_docker/shibboleth_idp/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Shibboleth IDP test container
+
+This container is a shibboleth SAML IdP designed to test SAML SPs (as tracim).
+
+## Configuration
+
+[//]: # FIXME - M.L - 2023/08/30
```
[the remark this answers is not in the document]
Comment (lebouquetin) [reply tools_docker/shibboleth_idp/README.md]: Reminder:

- FIXME must be fixed before merge into develop
- TODO may be merged (this is a non-blocking requirement)
- HACK / INFO are also mergeable

So, you should rename FIXME to TODO with initials and a date in ISO8601 style (2023-10-30, not with slashes which may be interpreted differently in french and american style dates) and add a one-liner explanation about the reason for this TODO

File: tools_docker/shibboleth_idp/README.md (hunk 2 above)
Comment (lebouquetin) [reply tools_docker/shibboleth_idp/README.md]: same remark: transform this into a TODO with a one line explanation and ISO8601 compatible format (with no ambiguity)

Comment (Millefeuille42) [reply tools_docker/shibboleth_idp/README.md]: Noted, fixed !

File: tools_docker/shibboleth_idp/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Millefeuille42) [reply tools_docker/shibboleth_idp/README.md]: Noted, fixed !

**Comment by Millefeuille42** [comment]
@lebouquetin Is this one mergeable ? Since the FIXME has been changed to TODOs

**Pull request merged**